verification/spec/connection-handshake/ConnectionHandshakeModule.tla

```diff
@@ -0,0 +1,353 @@
+--------------------- MODULE ConnectionHandshakeModule ---------------------
+
+(***************************************************************************
+
+    This module is part of the TLA+ specification for the
+    IBC Connection Handshake protocol (ICS3).
+    
+    This module captures the actions and operators of the ICS3 protocol.
+    Typically, it is an IBC module running on a chain that would implement
+    the logic in this TLA+ module, hence the name "ConnectionHandshakeModule"
+    sometimes abbreviated to "chModule" or "chm".
+
+    This module deals with a high-level spec of the ICS3 protocol, so it is
+    a simplification with respect to ICS3 proper in several regards:
+    - the modules assumes to run on a chain which we model as a simple
+    advancing height, plus a few more critical fields (see the 'store'),
+    but without any state (e.g., blockchain, transactions, consensus core);
+    - we model a single connection; establishing multiple connections is not
+    possible;
+    - we do not perform any cryptographic proofs or proof verifications.
+    - the abstractions we use are higher-level, and slightly different from
+    the ones in ICS3 (see e.g., ConnectionEnd and Connection records).
+    - the client colocated with the module is simplified, comprising only
+    a set of heights.
+
+ ***************************************************************************)
+
+EXTENDS Naturals, FiniteSets, Sequences, ICS3Types
+
+
+CONSTANTS MaxChainHeight,   \* Maximum height of the local chain.
+          ConnectionIDs,    \* The set of valid connection IDs.
+          ClientIDs,        \* The set of valid client IDs.
+          MaxBufLen         \* Maximum length of the input and output buffers.
+
+
+ASSUME Cardinality(ConnectionIDs) >= 1
+ASSUME Cardinality(ClientIDs) >= 1
+
+
+VARIABLES
+(******************************* Store *****************************
+     The store record of a chain contains the following fields:
+     
+     - id -- a string 
+       Stores the identifier of the chain where this module executes.
+
+     - latestHeight -- a Nat
+       Describes the current height of the chain.  
+
+     - connection -- a connection record.
+       Captures all the details of the connection on this chain.
+       For a full description of a connection record, see the
+       'Environment.Connections' set.
+
+     - client -- a client record.
+       Specifies the state of the client running on this chain.
```

Comment:
I think it would still be useful having "client record definition" here; even if it means duplicating text. The same for store. It simplifies reading functions after as you know what to expect there as fields.